cpp/src/Slice/Parser.cpp

```diff
@@ -522,47 +522,88 @@ Slice::DocComment::parseFrom(const ContainedPtr& p, DocLinkFormatter linkFormatt
 
         if (parseNamedCommentLine(l, paramTag, name, lineText))
         {
-            if (!isOperation)
+            if (!operationTarget)
             {
                 // If '@param' was put on anything other than an operation, ignore it and issue a warning.
-                string msg = "the '" + paramTag + "' tag is only valid on operations";
+                const string msg = "the '" + paramTag + "' tag is only valid on operations";
                 p->unit()->warning(p->file(), p->line(), InvalidComment, msg);
                 currentSection = nullptr;
             }
             else
             {
-                comment._parameters[name] = {};
-                currentSection = &(comment._parameters[name]);
-            }
-        }
-        else if (parseNamedCommentLine(l, throwsTag, name, lineText))
-        {
-            if (!isOperation)
-            {
-                // If '@throws' was put on anything other than an operation, ignore it and issue a warning.
-                string msg = "the '" + throwsTag + "' tag is only valid on operations";
-                p->unit()->warning(p->file(), p->line(), InvalidComment, msg);
-                currentSection = nullptr;
-            }
-            else
-            {
-                comment._exceptions[name] = {};
-                currentSection = &(comment._exceptions[name]);
+                // Check that the '@param <name>' corresponds to an actual parameter in the operation.
+                const ParameterList params = operationTarget->parameters();
+                const auto paramNameCheck = [&name](const ParameterPtr& param) { return param->name() == name; };
+                if (std::find_if(params.begin(), params.end(), paramNameCheck) == params.end())
```

Comment:
You should use std::none_of. I would also put the lambda inline, no need to give it a name.
https://en.cppreference.com/w/cpp/algorithm/all_any_none_of.

Reply:
Very nice!
I updated my `find_if`s accordingly.